lib/motions/general-motions.coffee

```diff
@@ -451,54 +451,50 @@ class MoveToMiddleOfScreen extends MoveToScreenLine
     height = lastScreenRow - firstScreenRow
     Math.floor(firstScreenRow + (height / 2))
 
-class ScrollKeepingCursor extends MoveToLine
-  previousFirstScreenRow: 0
-  currentFirstScreenRow: 0
+class ScrollKeepingCursor extends Motion
+  operatesLinewise: true
+  cursorRow: null
 
   constructor: (@editorElement, @vimState) ->
     super(@editorElement.getModel(), @vimState)
 
   select: (count, options) ->
-    finalDestination = @scrollScreen(count)
+    scrollTop = @scrollScreen(count)
     super(count, options)
-    @editorElement.setScrollTop(finalDestination)
+    @editorElement.setScrollTop(scrollTop)
 
   execute: (count) ->
-    finalDestination = @scrollScreen(count)
+    scrollTop = @scrollScreen(count)
     super(count)
-    @editorElement.setScrollTop(finalDestination)
+    @editorElement.setScrollTop(scrollTop)
 
-  moveCursor: (cursor, count=1) ->
-    cursor.setScreenPosition([@getDestinationRow(count), 0])
-
-  getDestinationRow: (count) ->
-    {row, column} = @editor.getCursorScreenPosition()
-    @currentFirstScreenRow - @previousFirstScreenRow + row
+  moveCursor: (cursor) ->
+    cursor.setScreenPosition(Point(@cursorRow, 0), autoscroll: false)
 
   scrollScreen: (count=1) ->
-    @previousFirstScreenRow = @editorElement.getFirstVisibleScreenRow()
-    destination = @scrollDestination(count)
-    @editorElement.setScrollTop(destination)
-    @currentFirstScreenRow = @editorElement.getFirstVisibleScreenRow()
-    destination
+    # TODO: We need to create an API for reading an editor's logical scroll
+    # position which may not yet have been flushed to the DOM. Currently, the
+    # only way to access this information is through this piece of private state.
+    currentScrollTop = @editorElement.component.presenter.pendingScrollTop ? @editorElement.getScrollTop()
```

Comment:
/cc @nathansobo @as-cii 

In this command, we're trying to read the editor's current scroll position, and update it accordingly. The problem is that if we run this command rapidly (e.g. holding down `ctrl-f`), calls to `TextEditorElement::getScrollTop()` will not reflect the value that was previously set via `::setScrollTop()`, because a DOM update hasn't happened yet.

Can you think of an easier way to do this? If not, maybe we need to add an API on `TextEditorElement` for accessing its current *desired* scroll position (i.e. the last value passed to `::setScrollTop`).